### PR TITLE
export default kibana client config

### PR DESCRIPTION
**Pull request by graphaelli**
## What does this PR do?

Exports `DefaultClientConfig`, to be used by beats framework users.

### CODE REVIEW

**Comment by graphaelli** [comment]
Test failure seems to be due to a flaky test:
```
21:10:11         --- FAIL: TestAll/kubernetes.state_deployment/ksm-v1.8.0.plain (12.85s)
21:10:11             testdata.go:272: Event was not expected: {"error":{"message":"decoding of metric family failed: net/http: request canceled (Client.Timeout exceeded while reading body)"},"event":{"dataset":"kubernetes.state_deployment","duration":115000,"module":"kubernetes"},"metricset":{"name":"state_deployment","period":10000},"service":{"address":"127.0.0.1:55555","type":"kubernetes"}}
```

**Comment by graphaelli** [comment]
on the travis side:
```
/usr/local/bin/docker-compose: line 1: !DOCTYPE: No such file or directory

 No such file or directory

/usr/local/bin/docker-compose: line 3: $'\r': command not found

/usr/local/bin/docker-compose: line 4: Hello: command not found

/usr/local/bin/docker-compose: line 6: $'\r': command not found

/usr/local/bin/docker-compose: line 84: syntax error near unexpected token `)'

` H��t��H���5�_ �%�_ @�%�_ h������%�_ h������%�_ h������%�_ h������%�_ h������%�_ h������%�_ h������%�_ h�p����%�_ �`����%�_ h	�P����%�_ h'

make: *** [build-image] Error 2

make: Leaving directory `/home/travis/gopath/src/github.com/elastic/beats/metricbeat'
```

**Comment by graphaelli** [comment]
Different tests failed (only on windows) after merging master back in - they seem to be unrelated to these changes.

File: libbeat/kibana/client_config.go [hunk 1]
```diff
@@ -36,8 +36,9 @@ type ClientConfig struct {
 	IgnoreVersion bool
 }
 
-var (
-	defaultClientConfig = ClientConfig{
+// DefaultClientConfig connects to a locally running kibana over HTTP
+func DefaultClientConfig() *ClientConfig {
```
Comment (urso) [inline libbeat/kibana/client_config.go]: Let's return default configs by value. We do so in most places (or should) that construct a default configuration. Just by judging from the function interface returning by value tells you that the value all belongs to you and can be freely modified without worries.

Comment (graphaelli) [reply libbeat/kibana/client_config.go]: sounds good, i'll make that change now

**Pull request merged**